### PR TITLE
Clean up docs/conf.py a bit

### PR DESCRIPTION
**Pull request by arcondello**
Remove some cruft, and do some minor reorganization. Highlight is that the minimum python version is now set algorithmically from the `setup.cfg`.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -1,45 +1,42 @@
...
+from dwaveoceansdk import __version__ as version
+from dwaveoceansdk import __version__ as release
+
+setup_cfg = configparser.ConfigParser()
+setup_cfg.read(os.path.join(sdk_directory, 'setup.cfg'))
+
+author = setup_cfg['metadata']['author']
+copyright = setup_cfg['metadata']['author']
+
+project = u'Ocean Documentation'
```
Comment (randomir) [inline docs/conf.py:33]: Unicode literal not needed in py3.

**Review by randomir — APPROVED** [review]
Nice job!

File: docs/conf.py [hunk 2]
```diff
@@ -58,65 +55,30 @@
...
 
-# The name of the Pygments (syntax highlighting) style to use.
 pygments_style = 'sphinx'
 
-# If true, `todo` and `todoList` produce output, else they produce nothing.
 todo_include_todos = True
 
 doctest_global_setup = """
-from __future__ import print_function, division
 import operator         # Used by dwave-binarycsp
```
Comment (JoelPasvolsky) [inline docs/conf.py:79]: No longer needed (dwave-binarycsp examples removed)

File: docs/conf.py [hunk 3]
```diff
@@ -148,179 +110,52 @@
...
-# Example configuration for intersphinx: refer to the Python standard library.
-#intersphinx_mapping = {'https://docs.python.org/': None,
-#                       'http://networkx.readthedocs.io/en/latest/': None}
-intersphinx_mapping = {'python': ('https://docs.python.org/3/', None),
+# -- Intersphinx ----------------------------------------------------------
+intersphinx_mapping = {
+    'python': ('https://docs.python.org/3/', None),
     'numpy': ('http://docs.scipy.org/doc/numpy/', None),
     'bson': ('https://api.mongodb.com/python/current/', None),
     'networkx': ('https://networkx.github.io/documentation/stable/', None),
```
Comment (JoelPasvolsky) [inline docs/conf.py:152]: I think this is now https://networkx.org/documentation/stable/

**Review by JoelPasvolsky — APPROVED** [review]
Much better. Thank you!

File: docs/conf.py (hunk 2 above)
Comment (arcondello) [reply docs/conf.py]: looks like we're still using it somewhere, so will restore it for now

**Pull request merged**